### PR TITLE
rework vue route

### PR DESCRIPTION
**Pull request by mshima**
https://github.com/jhipster/generator-jhipster/issues/17316 stack trace was backend related, but [only reproducible using vue frontend](https://github.com/jhipster/generator-jhipster/issues/17316#issuecomment-1000336462). The vue error was fixed by https://github.com/jhipster/generator-jhipster/pull/17410.

While migrating to vue-cli, a lot of errors happened due to unhandled routing exceptions.
Reimplement vue authentication handling moving it to the route `beforeEach` handler instead of mixed route and account service.

Fixes https://github.com/jhipster/generator-jhipster/issues/17316.

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/vue/templates/src/main/webapp/app/account/account.service.ts.ejs [hunk 1]
```diff
@@ -15,78 +13,68 @@ export default class AccountService {
...
-      }
-      return Promise.resolve(false);
+    const token = localStorage.getItem('<%=jhiPrefixDashed %>-authenticationToken') || sessionStorage.getItem('<%=jhiPrefixDashed %>-authenticationToken');
+    if (this.authenticated && this.userAuthorities && token) {
+      return;
+    }
+
+    const promise = this.retrieveAccount();
+    this.store.commit('authenticate', promise);
+    return await promise;
```
Comment (qmonmert) [inline generators/vue/templates/src/main/webapp/app/account/account.service.ts.ejs:72]: https://eslint.org/docs/latest/rules/no-return-await I think it's not a good practice

Comment (mshima) [inline generators/vue/templates/src/main/webapp/app/account/account.service.ts.ejs:72]:
```suggestion
    return promise;
```

File: generators/vue/templates/src/main/webapp/app/account/account.service.ts.ejs [hunk 2]
```diff
@@ -15,78 +13,68 @@ export default class AccountService {
...
-        this.store.commit('logout');
-        resolve(false);
-      });
-    });
+        return true;
+      }
+    } catch (error) {}
+
+    this.store.commit('logout');
+    return false;
```
[the remark this answers is not in the document]
Comment (mshima) [reply generators/vue/templates/src/main/webapp/app/account/account.service.ts.ejs:57]: Should be in the negate of `if (response.status === 200 && response.data) {` too.

File: generators/vue/templates/src/main/webapp/app/account/account.service.ts.ejs (hunk 1 above)
Comment (mshima) [reply generators/vue/templates/src/main/webapp/app/account/account.service.ts.ejs]: Leftover thanks..

File: generators/vue/templates/src/main/webapp/app/main.ts.ejs [hunk 3]
```diff
@@ -78,22 +82,60 @@ router.beforeEach(async (to, from, next) => {
...
+        // Ask for a new authentication
+<%_ if (authenticationTypeOauth2) { _%>
+        window.location.reload();
+<%_ } else { _%>
+        openLogin();
+<%_ } _%>
+        return;
+      }
+    }
+    console.log('Unauthorized!');
```
Comment (qmonmert) [inline generators/vue/templates/src/main/webapp/app/main.ts.ejs:130]: `console.log` to remove

File: generators/vue/templates/src/main/webapp/app/main.ts.ejs [hunk 4]
```diff
@@ -78,22 +82,60 @@ router.beforeEach(async (to, from, next) => {
...
+    return Promise.reject(error);
+  },
+  error => {
+    console.log('Server error!');
```
Comment (qmonmert) [inline generators/vue/templates/src/main/webapp/app/main.ts.ejs:134]: `console.log` to remove

**Review by qmonmert — APPROVED** [review]

File: generators/vue/templates/src/main/webapp/app/main.ts.ejs (hunk 3 above)
Comment (mshima) [reply generators/vue/templates/src/main/webapp/app/main.ts.ejs:130]:
```suggestion
```

File: generators/vue/templates/src/main/webapp/app/main.ts.ejs (hunk 4 above)
Comment (mshima) [reply generators/vue/templates/src/main/webapp/app/main.ts.ejs:134]:
```suggestion
```

**Review by DanielFran — APPROVED** [review]

**Pull request merged**